a1/src/main/java/ro/utcn/spet/a1/controller/ConsoleController.java

```diff
@@ -136,6 +152,50 @@ private void handleFilterTag(){
         questionService.findByTag(tag).forEach(s->print(s.toString()));
     }
 
+    private void handleAddAnswer(){
+        print("Question id: ");
+        int id=scanner.nextInt();
+        print("Write your answer: ");
+        String something=scanner.nextLine();
+        String text=scanner.nextLine();
+        Answer answer=answerService.addAnswer(text,this.user.getUsername(),id);
+        questionService.addAnswerToQuestion(id,answer);
```

Comment:
You should move the above two lines into the service layer. Creating the answer and adding it to the question should be an atomic operation.